pydicom/filereader.py

```diff
@@ -230,6 +233,15 @@ def data_element_generator(fp,
     element_struct_unpack = element_struct.unpack
     defer_size = size_in_bytes(defer_size)
 
+    tag_set = None
+    if specific_tags is not None:
+        tag_set = set()
```

Comment:
I want to make this as cheap as possible for the default case, so I first check for not None (which is the cheapest), then for containment (which in most cases will fail), and then for the Specific Character Set. I could make the tag_set which includes Specific Character Set before and use it, though, like:
``` python
if specific_tags is not None and tag not in extended_tag_set:
```